File: fastcampus/10816.py
import sys
input = sys.stdin.readline
from collections import defaultdict

n = int(input())
arr = list(map(int, input().split()))
m = int(input())
arr_b = list(map(int, input().split()))
card_count = defaultdict(int)

for el in arr:
    card_count[el] += 1

for el in arr_b:
    if el in card_count:
        print(card_count[el], end = ' ')
    else:
        print(0, end = ' ')

File: solved.ac/CLASS2_JS/10816.js
const fs = require("fs");
const inputPath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const stdin = fs
  .readFileSync(inputPath)
  .toString()
  .split("\n")
  .map((line) => line.trim());

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const solution = () => {
  const n = Number(input());
  const arr = input().split(" ").map(Number);

  const map = new Map();

  for (const el of arr) {
    const count = map.get(el) || 0;
    map.set(el, count + 1);
  }

  const m = Number(input());
  const arr2 = input().split(" ").map(Number);
  const ans = [];

  for (const el of arr2) {
    ans.push(map.get(el) || 0);
  }
  console.log(ans.join(" "));
};

solution();
